src/WebJobs.Script/Description/DotNet/FunctionSignature.cs

```diff
@@ -67,7 +67,7 @@ public string MethodName
             }
         }
 
-        private static bool AreParametersEquivalent(IParameterSymbol param1, IParameterSymbol param2)
+        private static bool AreParametersEquivalent(FunctionParameter param1, FunctionParameter param2)
```

Comment:
You could probably move this to an Equals overload on your new FunctionParameter class